test/integration/test_dataplane.go

```diff
@@ -204,29 +231,86 @@ func TestDataPlaneEssentials(t *testing.T) {
 
 	t.Log("verifying dataplane status is properly filled with backing service name and its addresses")
 	require.Eventually(t, testutils.DataPlaneHasServiceAndAddressesInStatus(t, GetCtx(), dataplaneName, clients), waitTime, tickTime)
+}
 
-	t.Log("updating dataplane spec with proxy service type of ClusterIP")
-	require.Eventually(t,
-		testutils.DataPlaneUpdateEventually(t, GetCtx(), dataplaneName, clients, func(dp *operatorv1beta1.DataPlane) {
-			dp.Spec.Network.Services.Ingress.Type = corev1.ServiceTypeClusterIP
-		}),
-		waitTime, tickTime)
+func TestDataPlaneServiceTypes(t *testing.T) {
+	t.Parallel()
+	namespace, cleaner := helpers.SetupTestEnv(t, GetCtx(), GetEnv())
 
-	t.Log("checking if dataplane proxy service type changes to ClusterIP")
-	require.Eventually(t, func() bool {
-		servicesClient := GetClients().K8sClient.CoreV1().Services(dataplane.Namespace)
-		dataplaneIngressService, err := servicesClient.Get(GetCtx(), dataplaneIngressService.Name, metav1.GetOptions{})
-		if err != nil {
-			t.Logf("error getting dataplane proxy service: %v", err)
-			return false
-		}
-		if dataplaneIngressService.Spec.Type != corev1.ServiceTypeClusterIP {
-			t.Logf("dataplane proxy service should be of ClusterIP type but is %s", dataplaneIngressService.Spec.Type)
-			return false
-		}
+	t.Log("deploying dataplane resource with 2 replicas")
+	dataplaneName := types.NamespacedName{
+		Namespace: namespace.Name,
+		Name:      uuid.NewString(),
+	}
+	dataplane := &operatorv1beta1.DataPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: dataplaneName.Namespace,
+			Name:      dataplaneName.Name,
+		},
+		Spec: operatorv1beta1.DataPlaneSpec{
+			DataPlaneOptions: operatorv1beta1.DataPlaneOptions{
+				Deployment: operatorv1beta1.DataPlaneDeploymentOptions{
+					DeploymentOptions: operatorv1beta1.DeploymentOptions{
+						Replicas: lo.ToPtr(int32(2)),
```

Comment:
Since this test only tests the service we can spawn just 1 replica (default) and save some resources from testing environment:

```suggestion
```